### PR TITLE
Match rfu_STC_setSendData_org in librfu_rfu.c

### PR DESCRIPTION
**Pull request by SleepyDweams**
For the past two weeks, I have been trying every possible combination to match this function, including modernizing and utilizing the ARM permuter at the very end.

## **Discord contact info**

Shoomer#2702

### CODE REVIEW

**Comment by PokeCodec** [comment]
Ready

**Comment by SleepyDweams** [comment]
Please check when you get the chance. This is library code that does not exist in any leak. Thank you!

File: include/librfu.h [hunk 1]
```diff
@@ -341,7 +341,7 @@ struct UNISend
                        //   Set with rfu_UNI_re_setSendData or rfu_UNI_readySendData, cleared when data is sent with rfu_REQ_sendData.
     u8 bmSlot;         // Expresses transmission destination slot as bits
     u16 payloadSize;   // Payload size of 1 transmission. "size" specified by rfu_UNI_setSendData or rfu_UNI_re_setSendData is used.
-    const void *src;   // Beginning address of transmission data
+    const u8 *src;   // Beginning address of transmission data
```
Comment (jiangzhengwenjz) [inline include/librfu.h:344]: breaks alignment of comments

File: src/librfu_rfu.c [hunk 2]
```diff
@@ -1409,10 +1410,16 @@ static u16 rfu_STC_setSendData_org(u8 ni_or_uni, u8 bmSendSlot, u8 subFrameSize,
...
         return ERR_SLOT_BUSY;
-    for (bm_slot_id = 0; bm_slot_id < RFU_CHILD_MAX && !((bmSendSlot >> bm_slot_id) & 1); ++bm_slot_id)
-        ;
+
+    for (bm_slot_id = 0; bm_slot_id < RFU_CHILD_MAX; ++bm_slot_id)
+    {
+        if ((bmSendSlot >> bm_slot_id) & 1)
+            break;
+    }
+
```
Comment (jiangzhengwenjz) [inline src/librfu_rfu.c:1422]: I'd be surprised if this actually changes codegen... Does it?

File: src/librfu_rfu.c [hunk 3]
```diff
@@ -1425,35 +1432,41 @@ static u16 rfu_STC_setSendData_org(u8 ni_or_uni, u8 bmSendSlot, u8 subFrameSize,
     sending = ni_or_uni & 0x20;
     if (sending || ni_or_uni == 0x40)
     {
-        u8 *dataType_p; // a hack to swap instructions
-
         slotStatus_NI = gRfuSlotStatusNI[bm_slot_id];
-        slotStatus_UNI = NULL;
+        slotStatus_UNI = NULL; // Needed to match. TODO: Is this actually needed
```
Comment (jiangzhengwenjz) [inline src/librfu_rfu.c:1436]: can be figured out with testing

**Comment by SleepyDweams** [comment]
Closed in favor of Kura's PR. 

Thank you for giving me the opportunity to work with you, even though I didn't deserve it. I hope you all luck in finishing the decomps.

Thank you also for all the lessons I learned while I was part of pret.

**Pull request closed without merging**